showcase/app/controllers/components.js

```
@@ -5,7 +5,7 @@
 
 import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
-import { scheduleOnce } from '@ember/runloop';
+import { later } from '@ember/runloop';
```

[question] why this change fixed the issue?

context: `scheduleOnce` is used in a lot of places in the showcase, for the replacement of mock data attributes to emulate states; why is not failing in other cases, and why it was not failing before? what is the change in your code that caused the failure of percy tests, the dynamic import? in which case, could this cause some side effect on the consumers that we are not thinking of?

---

well there is a multiple issues at play here:

* you are using controller which is already a bad idea for render based listeners and DOM manipulation as it's a singleton
* we a lying with `eslint-disable-next-line ember/no-runloop` if you look up what the reasoning for this rule is https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-runloop.md it would be obvious that is not aware of rendering async events like having ability to rerun when `tracked` property changes in our case we are swapping out actual component all together
* related to above since you are using `scheduleOnce` it only run once on the first paint but since we use `tracked` property to swap out component it doesn't rerun again

as for consumers with all my testing I have not seen anything failing in linked PRs

---

I was able to side step percy issue by initializing to an actual value instead of `null` which I think stops unwanted rerenders